pkgs/checks/lib/src/extensions/async.dart

```diff
@@ -26,6 +26,29 @@ extension FutureChecks<T> on Check<Future<T>> {
     });
   }
 
+  /// Expectst that the `Future` never completes as a value or an error.
+  ///
+  /// Immediately returns and does not cause the test to remain running if it
+  /// ends.
```

Comment:
I am a bit curious about how this doesn't end up making the test process hang? Are we relying on the VM knowing that a certain future can never be completed? What if some I/O related future never completed (maybe that isn't possible, but it seems so?).

Should this possibly take a duration and hold the test open for that long? Granted, generally I would discourage any such test but I don't really see the value in a test like this without something like that. It seems like it is asking for flaky tests...

Reply:
> I am a bit curious about how this doesn't end up making the test process hang?
> What if some I/O related future never completed (maybe that isn't possible, but it seems so?).

I don't think this should cause a test process to hang, but there may be a very narrow pattern where that is possible. This would need to be the only listener on that future.
This implementation won't behave any worse that the current `doesNotComplete`, and that hasn't caused us any issues.

https://github.com/dart-lang/test/blob/8235a25b98be448bca627d58b969de42b8b6633c/pkgs/test_api/lib/src/expect/future_matchers.dart#L101-L104

> Should this possibly take a duration and hold the test open for that long?

That might be a good idea. The current API in `test` does not have that behavior, but we could consider optional arguments. I'd prefer to hold that for a different PR.

Reply:
Ok, if we have an existing thing with this behavior, then just porting it directly sgtm